qdrant-landing/content/documentation/tutorials/retrieval-quality.md

```diff
@@ -117,111 +91,126 @@ client.upload_points(  # upload_points is available as of qdrant-client v1.7.1
 
 ```python
 def avg_precision_at_k(k: int):
     precisions = []
     for item in test_dataset:
+        # Perform ANN search
         ann_result = client.query_points(
             collection_name="arxiv-titles-instructorxl-embeddings",
             query=item["vector"],
             limit=k,
         ).points
     
+        # Perform exact search
         knn_result = client.query_points(
             collection_name="arxiv-titles-instructorxl-embeddings",
             query=item["vector"],
             limit=k,
             search_params=models.SearchParams(
-                exact=True,  # Turns on the exact search mode
+                exact=True,  # Enables exact search mode
             ),
         ).points
 
-        # We can calculate the precision@k by comparing the ids of the search results
-        ann_ids = set(item.id for item in ann_result)
-        knn_ids = set(item.id for item in knn_result)
+        # Calculate precision@k
+        ann_ids = set(point.id for point in ann_result)
+        knn_ids = set(point.id for point in knn_result)
         precision = len(ann_ids.intersection(knn_ids)) / k
         precisions.append(precision)
     
     return sum(precisions) / len(precisions)
 ```
 
-Calculating the `precision@5` is as simple as calling the function with the corresponding parameter:
+### Step 5: Measure Retrieval Quality
 
 ```python
-print(f"avg(precision@5) = {avg_precision_at_k(k=5)}")
+print(f"Initial avg(precision@5) = {avg_precision_at_k(k=5)}")
 ```
 
-Response:
+This step measures the initial retrieval quality before any tuning of the HNSW parameters. The HNSW (Hierarchical Navigable Small World) algorithm has two key parameters that influence search performance and quality:
 
-```text
-avg(precision@5) = 0.9935999999999995
-```
+- **m**: This parameter determines the maximum number of connections per node in the HNSW graph. A higher value for `m` increases the connectivity of the graph, potentially improving search accuracy at the cost of increased memory usage and indexing time. The default value for `m` is 16.
+- **ef_construct**: This parameter controls the size of the dynamic candidate list during index construction. A higher value of `ef_construct` leads to a more exhaustive search during the indexing phase, resulting in a higher quality graph and improved search accuracy. However, this comes at the cost of longer indexing times. The default value for `ef_construct` is 100.
+
+We will use the untuned HNSW as the baseline to compare how changes affect the precision of the search. Initially, we will use the default values of `m` (16) and `ef_construct` (100) for the HNSW algorithm. Later, we will double these values to observe their impact on retrieval quality.
```

Comment:
We have already written what the default values are, so we can shorten this sentence, like 
"We'll use the default `m` and `ef` as a baseline and then tweak the params to see how it affects the precision of the search."